### PR TITLE
OCPBUGS#32769: Adding dockercfg to list of types of secrets

### PR DESCRIPTION
**Pull request by bergerhoffer**
Version(s):
4.12+

Issue:
https://issues.redhat.com/browse/OCPBUGS-32769

Link to docs preview:
https://75082--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/pods/nodes-pods-secrets.html#nodes-pods-secrets-about-types_nodes-pods-secrets

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by bergerhoffer** [comment]
@sanchezl Can you please review to make sure this looks okay to you?

Preview: https://75082--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/pods/nodes-pods-secrets.html#nodes-pods-secrets-about-types_nodes-pods-secrets

**Review by sanchezl — APPROVED** [review]

**Comment by bergerhoffer** [comment]
@xingxingxia or @dpuniaredhat Can one of you please QE review this?

Preview: https://75082--ocpdocs-pr.netlify.app/openshift-enterprise/latest/nodes/pods/nodes-pods-secrets.html#nodes-pods-secrets-about-types_nodes-pods-secrets

**Review by jneczypor — COMMENTED** [review]
LGTM. I am still onboarding to peer review, so @michaelryanpeter will follow behind with his review. Thank you for your patience!

**Comment by michaelryanpeter** [comment]
LGTM

File: modules/nodes-pods-secrets-about.adoc [hunk 1]
```diff
@@ -64,10 +64,11 @@ which is the default.
 
 Specify one of the following types to trigger minimal server-side validation to ensure the presence of specific key names in the secret data:
 
-* `kubernetes.io/service-account-token`. Uses a service account token.
-* `kubernetes.io/basic-auth`. Use with Basic Authentication.
-* `kubernetes.io/ssh-auth`. Use with SSH Key Authentication.
-* `kubernetes.io/tls`. Use with TLS certificate authorities.
+* `kubernetes.io/basic-auth`: Use with Basic authentication
+* `kubernetes.io/dockercfg`: Use as an image pull secret
+* `kubernetes.io/service-account-token`: Uses a service account token
```
Comment (xingxingxia) [inline modules/nodes-pods-secrets-about.adoc:69]: Would you like to add both `kubernetes.io/dockercfg` and `kubernetes.io/dockerconfigjson`? This doc's other places mention `kubernetes.io/dockerconfigjson` examples. Upstream https://kubernetes.io/docs/concepts/configuration/secret/#secret-types has examples of both.

Comment (bergerhoffer) [reply modules/nodes-pods-secrets-about.adoc:69]: We definitely could. @sanchezl Do you know what we could put as the description for the dockerconfigjson one? Would it still be used for an image pull secret, but just a different file (`~/.docker/config.json`), or what?

Comment (sanchezl) [reply modules/nodes-pods-secrets-about.adoc:69]: @bergerhoffer The description would be exactly the same. The format of the file/data would be slightly different in the actual secret.

Comment (bergerhoffer) [reply modules/nodes-pods-secrets-about.adoc]: Thanks! I updated to add it with the same description. @xingxingxia @sanchezl if you can take another look and let me know if it's okay the way it is, or if the description needs adjusted to be different between the two.

File: modules/nodes-pods-secrets-about.adoc [hunk 2]
```diff
@@ -64,10 +64,12 @@ which is the default.
...
 Specify one of the following types to trigger minimal server-side validation to ensure the presence of specific key names in the secret data:
 
-* `kubernetes.io/service-account-token`. Uses a service account token.
-* `kubernetes.io/basic-auth`. Use with Basic Authentication.
-* `kubernetes.io/ssh-auth`. Use with SSH Key Authentication.
-* `kubernetes.io/tls`. Use with TLS certificate authorities.
+* `kubernetes.io/basic-auth`: Use with Basic authentication
+* `kubernetes.io/dockercfg`: Use as an image pull secret
+* `kubernetes.io/dockerconfigjson`: Use as an image pull secret
+* `kubernetes.io/service-account-token`: Uses a service account token
```
Comment (sanchezl) [inline modules/nodes-pods-secrets-about.adoc:70]:
```suggestion
* `kubernetes.io/service-account-token`: Use to obtain a legacy service account API token
```

**Review by sanchezl — APPROVED** [review]
/lgtm

**Comment by bergerhoffer** [comment]
@xingxingxia Mind giving this another quick look to make sure you're okay with the change luis suggested here? 

https://github.com/openshift/openshift-docs/pull/75082#discussion_r1595624162

**Pull request merged**